kubetest/aksengine_helpers.go

```diff
@@ -140,41 +140,51 @@ type KubernetesAddon struct {
+	Addons                           []KubernetesAddon     `json:"addons,omitempty"`
+	NetworkPolicy                    string                `json:"networkPolicy,omitempty"`
+	CloudProviderRateLimitQPS        float64               `json:"cloudProviderRateLimitQPS,omitempty"`
+	CloudProviderRateLimitBucket     int                   `json:"cloudProviderRateLimitBucket,omitempty"`
+	APIServerConfig                  map[string]string     `json:"apiServerConfig,omitempty"`
+	CloudControllerManagerConfig     map[string]string     `json:"cloudControllerManagerConfig,omitempty"`
+	KubernetesImageBase              string                `json:"kubernetesImageBase,omitempty"`
+	ControllerManagerConfig          map[string]string     `json:"controllerManagerConfig,omitempty"`
+	KubeletConfig                    map[string]string     `json:"kubeletConfig,omitempty"`
+	SchedulerConfig                  map[string]string     `json:"schedulerConfig,omitempty"`
+	KubeProxyMode                    string                `json:"kubeProxyMode,omitempty"`
+	LoadBalancerSku                  string                `json:"loadBalancerSku,omitempty"`
+	ExcludeMasterFromStandardLB      *bool                 `json:"excludeMasterFromStandardLB,omitempty"`
+	ServiceCidr                      string                `json:"serviceCidr,omitempty"`
+	DNSServiceIP                     string                `json:"dnsServiceIP,omitempty"`
+	OutboundRuleIdleTimeoutInMinutes int32                 `json:"outboundRuleIdleTimeoutInMinutes,omitempty"`
+	ClusterSubnet                    string                `json:"clusterSubnet,omitempty"`
+	CustomKubeAPIServerImage         string                `json:"customKubeAPIServerImage,omitempty"`
+	CustomKubeControllerManagerImage string                `json:"customKubeControllerManagerImage,omitempty"`
+	CustomKubeProxyImage             string                `json:"customKubeProxyImage,omitempty"`
+	CustomKubeSchedulerImage         string                `json:"customKubeSchedulerImage,omitempty"`
+	CustomKubeBinaryURL              string                `json:"customKubeBinaryURL,omitempty"`
+	UseManagedIdentity               *bool                 `json:"useManagedIdentity,omitempty"`
+	UserAssignedID                   string                `json:"userAssignedID,omitempty"`
+	WindowsContainerdURL             string                `json:"windowsContainerdURL,omitempty"`
+	WindowsSdnPluginURL              string                `json:"windowsSdnPluginURL,omitempty"`
+	Components                       []KubernetesComponent `json:"components,omitempty"`
+}
+
+// KubernetesComponent defines a component w/ configuration to include with the cluster deployment
+type KubernetesComponent struct {
+	Name       string                    `json:"name,omitempty"`
+	Enabled    *bool                     `json:"enabled,omitempty"`
+	Containers []KubernetesContainerSpec `json:"containers,omitempty"`
```

Comment:
does this also need to be specified?

Reply:
Yes, we need to parse the following new JSON fields I introduced in https://github.com/kubernetes-sigs/windows-testing/pull/267
```json
        "components": [
          {
            "name": "kube-addon-manager",
            "enabled": true,
            "containers": [
              {
                "name": "kube-addon-manager",
                "image": "mcr.microsoft.com/oss/kubernetes/kube-addon-manager:v9.1.5"
              }
            ]
          }
        ]
```

Reply:
Addressing kubetest-related change in a separate PR